cli/azd/cmd/util.go

```diff
@@ -283,7 +288,16 @@ func ensureEnvironmentInitialized(
 	}
 
 	if !hasPrincipalID {
-		principalID, err := azureutil.GetCurrentPrincipalId(ctx, userProfileService)
+		subscriptionId, found := env.Values[environment.SubscriptionIdEnvVarName]
```

Comment:
@ellismg @wbreza Quick thoughts how we should handle invariants like these? Should a `panic` be more appropriate to call this out? It would be impossible to get into this state given prior invariants (we check and set the `subId` prior to this.

A `panic` seems appropriate.

Reply:
This is fine for now but I see an opportunity here to refactor this environment management into its own component that can be responsible for all of this stuff.

Reply:
I agree that a panic would be better than an error since we know it would mean an invariant was violated.  However, to Wallace's later point, I would just call `GetSubscriptionId` here and just "know" it was going to return a good value, because of the invariants above and just not even bother with a panic.

Reply:
> it was going to return a good value, because of the invariants above a

I usually think what'd it happen in the future, if for any reason the code on the to changes, so it is not an invariant anymore to expect `subscriptionId` to be set... and when that change is done, it's hard to know it was an `invariant` for something below...  
Then, the code is now in a bad situation :)